Index: gray-streams.lisp
===================================================================
--- a/src/org/armedbear/lisp/gray-streams.lisp	(revision 15590)
+++ b/src/org/armedbear/lisp/gray-streams.lisp	(working copy)
@@ -246,6 +246,10 @@
 
 (defclass fundamental-binary-stream (fundamental-stream) ())
 
+(defmethod gray-stream-element-type ((s fundamental-binary-stream))
+  (declare (ignore s))
+  '(unsigned-byte 8))
+
 (defgeneric stream-read-byte (stream))
 (defgeneric stream-write-byte (stream integer))
 
